### PR TITLE
Add ClimaCore examples

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR adds the following ClimaCore examples from commit `bacefe624ca2d200ae8c5bcbdc91bbca6fd7fe75`:

```
examples/Manifest.toml
examples/Project.toml
examples/README.md
examples/common_spaces.jl
examples/implicit_solver_debugging_tools.jl
examples/ordinary_diff_eq_bug_fixes.jl
examples/hybrid/driver.jl
examples/hybrid/hyperdiffusion.jl
examples/hybrid/schur_complement_W.jl
examples/hybrid/staggered_nonhydrostatic_model.jl
examples/hybrid/box/bubble_3d_invariant_rhoe.jl
examples/hybrid/box/bubble_3d_invariant_rhotheta.jl
examples/hybrid/box/bubble_3d_rhotheta.jl
examples/hybrid/plane/density_current_2d_flux_form.jl
examples/hybrid/plane/density_current_2dinvariant_rhoe.jl
examples/hybrid/plane/density_current_2dinvariant_rhotheta.jl
examples/hybrid/sphere/README.md
examples/hybrid/sphere/baroclinic_wave_rhoe.jl
examples/hybrid/sphere/baroclinic_wave_rhotheta.jl
examples/hybrid/sphere/baroclinic_wave_utilities.jl
examples/hybrid/sphere/held_suarez_rhoe.jl
examples/hybrid/sphere/held_suarez_rhoe_int.jl
examples/hybrid/sphere/held_suarez_rhotheta.jl
examples/hybrid/sphere/held_suarez_rhotheta_tempest.jl
examples/hybrid/sphere/remap_pipeline.jl
examples/hybrid/sphere/solid_body_rotation_3d.jl
```

This was performed by

```
mkdir examples/hybrid
mkdir examples/hybrid/box
mkdir examples/hybrid/plane
mkdir examples/hybrid/sphere
# then, for all files:
cp ../ClimaCore.jl/filename filename
```

As a sanity check, `git diff --name-only` yields
```
.buildkite/pipeline.yml
Project.toml
examples/Project.toml
examples/README.md
examples/common_spaces.jl
examples/hybrid/box/bubble_3d_invariant_rhoe.jl
examples/hybrid/box/bubble_3d_invariant_rhotheta.jl
examples/hybrid/box/bubble_3d_rhotheta.jl
examples/hybrid/driver.jl
examples/hybrid/hyperdiffusion.jl
examples/hybrid/plane/density_current_2d_flux_form.jl
examples/hybrid/plane/density_current_2dinvariant_rhoe.jl
examples/hybrid/plane/density_current_2dinvariant_rhotheta.jl
examples/hybrid/schur_complement_W.jl
examples/hybrid/sphere/README.md
examples/hybrid/sphere/baroclinic_wave_rhoe.jl
examples/hybrid/sphere/baroclinic_wave_rhotheta.jl
examples/hybrid/sphere/baroclinic_wave_utilities.jl
examples/hybrid/sphere/held_suarez_rhoe.jl
examples/hybrid/sphere/held_suarez_rhoe_int.jl
examples/hybrid/sphere/held_suarez_rhotheta.jl
examples/hybrid/sphere/held_suarez_rhotheta_tempest.jl
examples/hybrid/sphere/remap_pipeline.jl
examples/hybrid/sphere/solid_body_rotation_3d.jl
examples/hybrid/staggered_nonhydrostatic_model.jl
examples/implicit_solver_debugging_tools.jl
examples/ordinary_diff_eq_bug_fixes.jl
```

### CODE REVIEW

**Comment by bischtob** [comment]
I'd like to have a solid round of cleanup after this PR too :)

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -95,3 +95,89 @@ steps:
...
+          TEST_NAME: "sphere/baroclinic_wave_rhoe"
+          CLIMACORE_DISTRIBUTED: "MPI"
+        agents:
+          slurm_ntasks: 2
+
+      - label: ":computer: baroclinic wave (ρθ)"
+        command: "julia --color=yes --project=examples examples/hybrid/driver.jl"
+        artifact_paths: "examples/hybrid/sphere/output/baroclinic_wave_rhotheta/Float32/*"
+        env:
+          TEST_NAME: "sphere/baroclinic_wave_rhotheta"
```
Comment (jiahe23) [inline .buildkite/pipeline.yml:159]: let's leave one line below this

**Review by jiahe23 — APPROVED** [review]
just one comment in `pipeline.yml`, otherwise it looks good

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
Bingo.

**Comment by charleskawczynski** [comment]
Bors r+

**Pull request merged**